src/lib/model.ts

```diff
@@ -287,23 +287,30 @@ export class Event {
 			capacity = data.capacity
 		}
 
-		return new Event (date, location, capacity, availability, data.id)
+		let status = 'Active'
+		if (data.status) {
+			status = data.status
+		}
```

Comment:
Could refine this a bit with a ternary:

```suggestion
		 status = data.status ? data.status : 'Active'
```